Lib/test/test_typing.py

```diff
@@ -7586,6 +7586,17 @@ def test_total(self):
         self.assertEqual(Options.__required_keys__, frozenset())
         self.assertEqual(Options.__optional_keys__, {'log_level', 'log_path'})
 
+    def test_total_inherits_non_total(self):
+        class TD1(TypedDict, total=False):
+            a: int
+
+        self.assertIs(TD1.__total__, False)
+
+        class TD2(TD1):
+            b: str
+
+        self.assertIs(TD2.__total__, True)
```

Comment:
I know you're not changing behaviour here, but are the precise semantics here specified anywhere? Mypy has some interesting behaviour here -- it treats all the keys present in `TD2` as required, but all the keys present in `TD1` as not required. That implies to me that `TD2` should _not_ be seen as a "total" `TypedDict`.

https://mypy-play.net/?mypy=latest&python=3.11&gist=e7ef88dd8c2b5697297c4739d471ac45

The `__total__` attribute seems to be pretty meaningless these days, though, in a post-PEP-655 world:

```pycon
>>> from typing import *
>>> class Foo(TypedDict):
...     x: Required[int]
...     y: NotRequired[str]
...
>>> Foo.__total__
True
```

Maybe we should treat that as a bug and make it meaningful. Or maybe we should just be clearer in the documentation that `__total__` doesn't indicate whether or not all keys in the `TypedDict` are required or not, it just literally indicates whether that specific `TypedDict` was constructed using `total=True` (but I'm not sure why that would be useful to anybody).

Reply:
The mypy behavior is right; that was the way to mix Required and NotRequired keys before PEP 655.

Agree that `__total__` isn't that useful and you should generally look at `__required_keys__` and `__optional_keys__`. However, I think the current behavior is right. If you do want to use `__total__` for introspection, you should look not just at the current class's attribute, but also those of base classes.

Reply:
Submitted #109547 with some enhancements to the docs.